### PR TITLE
use_dpctl_round_func_in_dpnp

### PR DESCRIPTION
**Pull request by vtavana**
The PR changes the implementation of `dpnp.round` and its alias `dpnp.around`. These functions now invoke `oneapi::mkl::vm::int` from pybind11 extension of OneMKL VM if possible or uses `dpctl.tensor.round` implementation if not.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: tests/third_party/cupy/math_tests/test_rounding.py [hunk 1]
```diff
@@ -70,8 +71,8 @@ def test_around(self):
         self.check_unary("around")
         self.check_unary_complex("around")
 
-    def test_round_(self):
-        self.check_unary("round_")
+    def test_round(self):
+        self.check_unary("round")
         self.check_unary_complex("around")
```
Comment (antonwolfy) [inline tests/third_party/cupy/math_tests/test_rounding.py:78]: will it be possible to modify test and to remove the following lines:
```
tests/third_party/cupy/math_tests/test_rounding.py::TestRound_param_0_{decimals=-2}::test_round_out
tests/third_party/cupy/math_tests/test_rounding.py::TestRound_param_1_{decimals=-1}::test_round_out
tests/third_party/cupy/math_tests/test_rounding.py::TestRound_param_2_{decimals=0}::test_round_out
tests/third_party/cupy/math_tests/test_rounding.py::TestRound_param_3_{decimals=1}::test_round_out
tests/third_party/cupy/math_tests/test_rounding.py::TestRound_param_4_{decimals=2}::test_round_out
```
from `tests/skipped_tests_gpu_no_fp64.tbl`

File: tests/third_party/cupy/math_tests/test_rounding.py [hunk 2]
```diff
@@ -115,6 +116,7 @@ def test_round_out(self, xp):
         }
     )
 )
+@pytest.mark.usefixtures("allow_fall_back_on_numpy")
```
[the remark this answers is not in the document]
Comment (vtavana) [reply tests/third_party/cupy/math_tests/test_rounding.py:119]: Instead, I modified the test to work with or without support for double-precision floating type.

File: dpnp/dpnp_iface_mathematical.py [hunk 3]
```diff
@@ -1771,19 +1756,43 @@ def remainder(
     )
 
 
-def round_(a, decimals=0, out=None):
+def round(x, decimals=0, out=None):
```
[the remark this answers is not in the document]
Comment (vtavana) [reply dpnp/dpnp_iface_mathematical.py:1759]: will address in a separate PR

**Review by antonwolfy — APPROVED** [review]
Great! Thank you @vtavana

**Pull request merged**